homework-5/homework-5.py

```diff
@@ -0,0 +1,86 @@
+# Backend Programming with Python
+# Homework 5 - Battleships
+
+import random, math
+
+
+class SpaceShip:
+    def __init__(self, name, hp, armour, damage):
+        self.name = name
```

Comment:
Violation of encapsulation - name, hp, armour and damage can all be made private. If you really want to see these details from the outside (without allowing change), just create some accessors for them - properties or getter methods, up to you.